pkg/coremodel/pluginmeta/coremodel.cue

```diff
@@ -131,6 +131,35 @@ seqs: [
 				// in all orgs
 				autoEnabled?: bool
 
+				// List of RBAC roles and their assignments.
+				roles?: [...#RoleRegistration]
+
+				// RoleRegistration describes the definition of an RBAC role
+				// and its assignments to BasicRoles
+				#RoleRegistration: {
+					// RBAC role definition
+					role: #Role
+
+					// Assignment to a basic role 
+					// (Viewer, Editor, Admin, Grafana Admin)
+					grants: [...#BasicRole]
+				}
+
+				// Role describes an RBAC role and its permissions
+				#Role: {
+					name: string,
+					displayName: string,
+					description: string,
```

Comment:
nit: these (display name and description) are optional, right?

Reply:
My thinking was to make them mandatory as it's low effort on the developer side to specify them but brings UX value.
If we were to allow the role name not to be specified, we'd be relying on the [`fallbackDisplayName`](https://github.com/grafana/grafana/blob/f76ba90078fa85c8b434e7f660812ca714353d8d/pkg/services/accesscontrol/models.go#L161) which currently only handles `fixed` roles, I'd need to modify to work with plugin roles.

Reply:
We could however go one step further and remove the role name to compute it from the display name and the plugin ID.

Reply:
Changes done in https://github.com/grafana/grafana/pull/58166 :)

Reply:
Smart! 🧠